### ISSUE TITLE
Csla 4.8.1 and PropertyRule class

### ISSUE THREAD

**Issue by Chicagoan2016** [MarimerLLC/cslaforum#651]
I am implementing a simple business rule in Csla 4.8.1 and inheriting the rule from 'PropertyRule'. The compiler is complaining about the 'Execute' method. "No suitable method found to override".
The signature of the method is
protected override void Execute(RuleContext context)

But If change the parameter type from 'RuleContext' to IRuleContext, the compiler is fine with it.
I know that RuleContext in Csla 4.8.1 implments IRuleContext interface. I wonder why is the compile time error?

Kind Regards

**Version and Platform**
CSLA version: 4.7.100
OS: Windows, Linux, iOS, Android, etc.
Platform: WinForms, WPF, ASP.NET Core, MVC, Xamarin, etc.

**Comment by rockfordlhotka** [comment]
The method signature was changed so it now requires an `IRuleContext`.

**Comment by kcabral817** [comment]
I ran into the same issue.  What I did was to right on the inherited item " : BusinessRule" which will show you the signature of the object.  If you look down in the protected virtual void Execute signature you'll see "(IRuleContext context)".  This is the signature that you're looking for.  As Rocky said the signature was changed from RuleContext to IRuleContext.  So any time you use the protected override void Execute(RuleContext context) method you'll need to change the RuleContext to IRuleContext.  This is the easiest and fastest way to find what the class is doing along with its method signatures.  I hope this helps.
